test/python/picongpu/quick/picmi/grid.py

```diff
@@ -84,3 +84,120 @@ def test_n_gpus_wrong_numbers(self):
                     picongpu_n_gpus=not_ngpus_dist,
                 )
                 grid.get_as_pypicongpu()
+
+    def test_supercell(self):
+        """test supercell default"""
+        grid = picmi.Cartesian3DGrid(
+            number_of_cells=[192, 2048, 12],
+            lower_bound=[0, 0, 0],
+            upper_bound=[3.40992e-5, 9.07264e-5, 2.1312e-6],
+            lower_boundary_conditions=["open", "open", "periodic"],
+            upper_boundary_conditions=["open", "open", "periodic"],
+            picongpu_n_gpus=[1, 1, 1],
+            picongpu_super_cell_size=(8, 8, 4),
+        )
+        g = grid.get_as_pypicongpu()
+        assert g.super_cell_size == (8, 8, 4), "supercell should be [8,8,4]"
+
+    def test_super_cell_mismatch_no_dist(self):
+        with self.assertRaisesRegex(Exception, ".*GPU- and/or super-cell-distribution.*"):
+            grid = picmi.Cartesian3DGrid(
+                number_of_cells=[192, 2048, 12],
+                lower_bound=[0, 0, 0],
+                upper_bound=[3.40992e-5, 9.07264e-5, 2.1312e-6],
+                lower_boundary_conditions=["open", "open", "periodic"],
+                upper_boundary_conditions=["open", "open", "periodic"],
+                picongpu_super_cell_size=(7, 8, 4),
+            )
```

Comment:
Untested but you get the idea:
```suggestion
        number_of_cells = [192, 2048, 12]
        super_cell_size = [7, 8, 4]
        # pre-condition for a mismatch
        assert any(np.asarray(number_of_cells) % super_cell_size != 0)
        with self.assertRaisesRegex(Exception, ".*GPU- and/or super-cell-distribution.*"):
            grid = picmi.Cartesian3DGrid(
                number_of_cells=number_of_cells,
                lower_bound=[0, 0, 0],
                upper_bound=[3.40992e-5, 9.07264e-5, 2.1312e-6],
                lower_boundary_conditions=["open", "open", "periodic"],
                upper_boundary_conditions=["open", "open", "periodic"],
                picongpu_super_cell_size=super_cell_size,
            )
```

This is non-blocking and maybe personal style but I like to be explicit about what's the aspects to focus on in a test. If you like this approach, you'll surely figure out how to apply it to the other tests yourself.

Reply:
You will have to take your beef to ChatGPT, I was 100% vibe coding the tests :D, but I agree hahaha

Reply:
I will allow myself to skip this for now :D

Reply:
Okay.